core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -228,6 +235,54 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 
 	logger.Debugw(fmt.Sprintf("EthConfirmer: fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
+	attemptsByAddress := make(map[gethCommon.Address][]models.EthTxAttempt)
+	for _, att := range attempts {
+		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+	}
+
+	for from, attempts := range attemptsByAddress {
+		latestBlockNonce, err := ec.getNonceForLatestBlock(ctx, from)
+		logger.Debugw(fmt.Sprintf("EthConfirmer: There are %v attempts from address %v, and latest nonce is %v",
+			len(attempts), from, latestBlockNonce), "blockNum", blockNum)
+
+		if err != nil {
+			return errors.Wrapf(err, "unable to fetch pending nonce for address: %v", from)
+		}
+
+		likelyConfirmed := make([]models.EthTxAttempt, 0)
```

Comment:
Do we need to `make` it if we assign it below?

Reply:
yes because it may be never assigned in that loop, but i'll change it a bit